File: src/ruby/Ecosistema/Biosenosis.rb
require('./Ecosistema/Locacion.rb');
require('./Ecosistema/Biotopo.rb');

module Ecosistema
	public
	class Biosenosis

		# @ReturnType String
		public
		def getNombre()
			return @_nombre
		end

		# @ParamType aNombre String
		# @ReturnType void
		public
		def setNombre(aNombre)
			@_nombre = aNombre
		end

		# @ReturnType ecosistema.Locacion
		public
		def getLocacion()
			return @_locacion
		end

		# @ParamType aLocacion ecosistema.Locacion
		# @ReturnType void
		public
		def setLocacion(aLocacion)
			@_locacion = aLocacion
		end

		public
		def initialize()
			# @AttributeType String
			@_nombre
			# @AttributeType ecosistema.Locacion
			@_locacion
			# @AttributeType ecosistema.Biotopo
			# # @AssociationType ecosistema.Biotopo
			@_atLocation_location___Location_
			# @AttributeType ecosistema.Locacion
			# # @AssociationType ecosistema.Locacion
			# # @AssociationMultiplicity 1
			# # @AssociationKind Aggregation
			@_laLocacion

			# Not yet implemented
		end
	end
end

File: src/python/ecosistema/Biosenosis.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
from ecosistema import Locacion
from ecosistema import Biotopo
from typing import List

class Biosenosis(object):
	def getNombre(self) -> String:
		return self.___nombre

	def setNombre(self, aNombre : String) -> None:
		self.___nombre = aNombre

	def getLocacion(self) -> Locacion:
		return self.___locacion

	def setLocacion(self, aLocacion : Locacion) -> None:
		self.___locacion = aLocacion

	def __init__(self):
		self.___nombre : String = None
		self.___locacion : Locacion = None
		self._atLocation_location___Location_ : Biotopo = None
		self._laLocacion : Locacion = None
		"""# @AssociationMultiplicity 1
		# @AssociationKind Aggregation"""

